### PR TITLE
update clap to 3.0

### PR DESCRIPTION
**Pull request by bluejekyll**
closes: #1614

depends on: https://github.com/clap-rs/clap/issues/3267

### CODE REVIEW

File: bin/Cargo.toml [hunk 1]
```diff
@@ -71,7 +71,7 @@ name = "named"
 path = "src/named.rs"
 
 [dependencies]
-clap = "2.33"
+clap = {version = "3.0", features = ["cargo"] }
```
Comment (djc) [inline bin/Cargo.toml:74]: Spaces inside braces for consistency?

**Review by djc — APPROVED** [review]

File: bin/Cargo.toml (hunk 1 above)
Comment (bluejekyll) [reply bin/Cargo.toml:74]: Probably just a mistake. I’ll update. I’m currently fighting clap on a minimum rust version.

might need to update to 1.52 if we want this, but maybe even more recent. I’ll know in a minute.

Comment (djc) [reply bin/Cargo.toml:74]: clap is 1.54, but I filed an issue upstream, should be easy to get it down to 1.52.

**Comment by bluejekyll** [comment]
I just bumped to 1.54, that was released in July... by the time we publish the next version, I think that will be 6 months ago, so I think that's reasonable. Thoughts?

**Comment by bluejekyll** [comment]
> I just bumped to 1.54, that was released in July... by the time we publish the next version, I think that will be 6 months ago, so I think that's reasonable. Thoughts?

@djc, thoughts on this?

**Comment by djc** [comment]
I'd probably still prefer to be a little more conservative, but I'm also fine if you prefer to go with 1.54.

**Comment by bluejekyll** [comment]
My sense is that 6 months should be reasonable. I was checking out Tokio, and they will only guarantee 4 1/2 months (3 releases). I think we should make an explicit policy. That being said Tokio is still back on 1.4x...

So if you're ok with it, I think we'll bump the version, and thus unblock this and get another alpha out.

**Comment by djc** [comment]
Sounds good!

**Pull request merged**